numpy/core/src/umath/ufunc_object.c

```diff
@@ -2124,18 +2124,27 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
 
     if (axes) {
         /*
-         * possibly remap axes.
+         * possibly remap axes, using newly allocated memory.
          */
+        remap_axis = (npy_intp *)PyArray_malloc(NPY_SIZEOF_INTP * nop);
+        remap_axis_memory = (int *)PyArray_malloc(sizeof(int) * nop *
+                                                  NPY_MAXDIMS);
+        if (remap_axis == NULL || remap_axis_memory == NULL) {
```

Comment:
Memory leak if one is allocated but the other is not.

Reply:
I free both in the `fail` part, so I think this should be OK, no?

Reply:
You're right, I forgot about that.